packages/core/src/parser/Domain.ne

```diff
@@ -82,11 +83,15 @@ type_decl -> "type" __ identifier (_ "(" _ type_params _ ")"):? (_ "<:" _ sepBy1
   }
 %}
 
-predicate -> "predicate" __ identifier type_params_list args_list {%
-  ([kw, , name, params, args]): PredicateDecl<C> => ({
+# This now works with symmetric predicate declarations.
+predicate -> ("symmetric" __):? "predicate" __ identifier type_params_list args_list {%
+  ([sym, kw, , name, params, args]): PredicateDecl<C> => ({
     ...nodeData,
     ...rangeFrom([rangeOf(kw), ...args, ...params]),
```

Comment:
Minor point: if qualifiers like `symmetric` come before `predicate`, the starting source position will need to change. Now that I think about it, I kind of like `predicate symmetric` a bit more? Not sure if the concrete syntax was discussed at all

Reply:
Yep that seems right!

Reply:
I just pushed some new code if you want to take a look!